### PR TITLE
[#70] get username from request is more flexible

### PR DESCRIPTION
**Pull request by shnela**
- Every function which loads username from request has the optional paramter: get_username
piggyback:
- removed redefinition of USERNAME_FORM_FIELD from defender.config

### CODE REVIEW

File: defender/config.py [hunk 1]
```diff
@@ -54,9 +54,6 @@ def get_setting(variable, default=None):
 ERROR_MESSAGE = ugettext_lazy("Please enter a correct username and password. "
                               "Note that both fields are case-sensitive.")
 
-# use a specific username field to retrieve from login POST data
-USERNAME_FORM_FIELD = get_setting('DEFENDER_USERNAME_FORM_FIELD', 'username')
-
```
Comment (kencochrane) [inline defender/config.py]: good catch

**Comment by kencochrane** [comment]
@shnela thank you for the PR. Since you have a pretty good use case for using with DRF, maybe it makes sense to update the README with and example of how you are using it, so that others can do the same?

Once the tests pass, I can merge.

**Comment by kencochrane** [comment]
ok @coveralls I get the picture, you don't need say it 9 times.

**Comment by kencochrane** [comment]
ok, all tests passed.

**Comment by shnela** [comment]
@kencochrane yes, I will update README with simple example of `BasicAuthentication` for `rest_framework` tomorrow.

**Pull request merged**